Guide: Add some links to the suggested dev env

danielarai-ms opened the pull request:
Add a link to the instructions for installing the Windows SDK, and add a link to installing the Windows Rust toolchain in the Windows deps section of the suggested dev environment.

daprilik commented on Guide/src/dev_guide/getting_started/suggested_dev_env.md:
```diff
@@ -236,8 +236,9 @@ Note that this requires some additional dependencies, described below.
-Visual Studio build tools must be installed, along with the Windows SDK. This is
-the same as what's required to build OpenVMM on windows.
+Visual Studio build tools must be installed, along with the
+[Windows SDK](https://developer.microsoft.com/en-us/windows/downloads/windows-sdk/).
```
I would prefer if we skipped adding this extra link, and only linked back to the Windows instructions earlier in the guide. That way, we have a single source of truth to update, if/when this process is changed/refined.

Also, please feel free to rewrite this blurb to flow better / be a bit clearer, if you think that'd be useful.

danielarai-ms replied:
I'm all for a single source of truth, but this is the only place in the entire guide that I found that references the Windows SDK. Should I add the instructions for the Windows SDK to the section in the guide under Developer Guide -> Getting Started -> On Windows and remove the reference to the SDK here?

daprilik replied:
Ah, interesting. You're right that https://openvmm.dev/guide/dev_guide/getting_started/windows.html#installing-rust doesn't actually say you must install the Windows SDK.

Lets make sure we're giving the right advice here.

Are we sure the Windows SDK is actually necessary, or are the existing "Installing Rust on Windows" instructions sufficient?

If the current instructions work, lets remove the reference to needing the Windows SDK from this paragraph. If it turns out those instructions are incomplete, and the Windows SDK is required, lets update those instructions to also have folks install the Windows SDK.

jstarks replied:
Visual Studio installs the Windows SDK by default.

danielarai-ms replied:
I had both Visual Studio and vscode installed on my system, but I did not have the Windows SDK installed. I don't know what sequence of operations got me into that state, but it's definitely possible to have Visual Studio without the Windows SDK.

The Windows SDK is required when cross-compiling at least some targets from WSL2 to Windows. The cross_tool.py script queries the HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Windows Kits\Installed Roots registry key and uses tools that I assume are part of the Windows SDK. The instructions under VMM Tests didn't work for me until I installed the Windows SDK.

(Also, sort of unrelated, I don't see anything in the section on VMM Tests that says it's using Windows to run the tests. Would those instructions work if you were running them from actual Linux and not WSL2?)

There are instructions for setting up the cross-compilation environment. They're under Developer Environment -> Suggested Dev Environment -> [WSL2] Cross Compiling from WSL2 to Windows. This is the only place I found the Windows SDK listed as a requirement. Since this section says it's "Suggested", I ignored most of this section the first time through. Possibly the cross-compilation instructions deserve to be separated out from the "Suggested" section.

danielarai-ms commented:
I have updated this pull request. I removed the link the Windows SDK instructions from the cross-compliation section. Instead, I added some more details about the Windows SDK to the Windows setup page. I also added `RUST_BACKTRACE` to the list of suggested variables for `WSLENV`. (That's unrelated to the Windows SDK but was helpful to me.)

danielarai-ms commented:
@microsoft-github-policy-service agree company="Microsoft"

tjones60 commented on Guide/src/dev_guide/getting_started/windows.md:
````diff
@@ -47,6 +49,11 @@ Or, you can install the tool via the powershell command below.
 PS> winget install Microsoft.VisualStudio.2022.Community --override "--quiet --add Microsoft.VisualStudio.Component.VC.Tools.x86.x64"
 ```
 
+The Windows SDK must also be installed. You will typically get this automatically when installing
````
I noticed this was a dependency when setting up the self hosted test runners, but forgot to update the guide. We should move this paragraph above the `winget` command and fix the `winget` command to include `--add Microsoft.VisualStudio.Component.Windows11SDK.22621` (or the equivalent for the latest package). I also found that I needed `--add Microsoft.VisualStudio.Component.VC.Llvm.Clang` for compiling on ARM, and "C:\Program Files (x86)\Microsoft Visual Studio\2022\BuildTools\VC\Tools\Llvm\ARM64\bin" needed to be added to path.

danielarai-ms replied:
I don't think there's an alias to install the latest version of the Windows SDK the way there is for the C++ toolchain, or at least I couldn't find it. I found a list of installable components here: https://learn.microsoft.com/en-us/visualstudio/install/workload-component-id-vs-build-tools?view=vs-2022

I am unfortunately unable to verify the winget instructions. If I uninstall the C++ development tools and Windows SDK via the Visual Studio Installer GUI, then when I run the winget command, it just says "Startingpackage install... Successfully installed", but it doesn't actually do anything. I have to reinstall via the GUI to get it to do anything.

I don't have an ARM build machine, so maybe someone who has one can verify specifically what needs to be installed and make an additional guide update for ARM? I'm guessing that the Microsoft.VisualStudio.Component.VC.Tools.x86.x64 component probably isn't needed for ARM.

I've reorganized with Trevor's suggestion, except for the ARM updates, since I can't test those.

tjones60 replied:
Thanks, this looks good. I can verify the ARM stuff and update that at a later date. Unfortunately, when installing Visual Studio via `winget`, the command will complete before the installation is finished. You can monitor task manager to see if the Visual Studio installer is still running. In my experience, it does complete successfully if you wait for the process to disappear.

smmalis37 commented on Guide/src/dev_guide/getting_started/windows.md:
```diff
@@ -33,18 +33,27 @@ over to it.
 -> `MSVC v143 - VS 2022 C++ x64/x86 build tools (latest)`.
 
+The Windows SDK can be installed via `Visual Studio Installer` -> `Modify` -> `Individual Components`
+-> `Windows 11 SDK (10.0.26100.0)`.
```
We should have some guidance here about different version numbers, since new versions come out occasionally. It's probably fine to just recommend whatever the biggest number is.

danielarai-ms replied:
Lines 41-43 say "Windows 11 SDK version 26100 is the latest
version as of this update, but you should install the newest version if there is a newer
version."
